src/osd/OSDMap.cc

```diff
@@ -2849,8 +2849,34 @@ void OSDMap::print(ostream& out) const
 class OSDTreePlainDumper : public CrushTreeDumper::Dumper<TextTable> {
 public:
   typedef CrushTreeDumper::Dumper<TextTable> Parent;
-  OSDTreePlainDumper(const CrushWrapper *crush, const OSDMap *osdmap_)
-    : Parent(crush), osdmap(osdmap_) {}
+
+  bool down = false;
+  bool up = false;
+  bool in = false;
+  bool out = false;
+
+  OSDTreePlainDumper(const CrushWrapper *crush, const OSDMap *osdmap_,
+		     const string& state)
+    : Parent(crush), osdmap(osdmap_) {
+    if (state.find("up") != std::string::npos)
+      up = true;
+    else if (state.find("down") != std::string::npos)
```

Comment:
It would be even more useful if it would be possible to combine flags (like "in+down"). Right now it accepts only one flag.